File: ejercicio3.py

def primus(x): #funcion que determina si un numero es primo 
        i=1
        while i<x-1:
            i=i+1
            sob=x%i
    
            if (sob==0):
                return "" #retorna un valor "vacio" si el numero evaluado no es primo
            if (i==x-1):
                return x    #retorna solo numero primos

x=int (input("Digite inicio de rango: "))   #solicitud de valores 
y=int (input("Digite final de rango: "))
aux=x-1
while(aux<y): #ciclo que se repite mientras se evalua si los numeros dentro de un rango son primos o no 
    aux=aux+1
    ret=primus(aux)
    if (aux==x): #impresion unica dentro de un ciclo
        print ("Los numero primos en el rango indicado son:")
    print (ret,end=" ") #imprime todos los numeros primos dentro del rango asignado

File: ejercicio3.cpp
#include <iostream>
#include <stdlib.h>
using namespace std;
//Funcion que encuentra si un numero es primo o no 
int primus (int x){
    int i = 1;
    int sob = 0;
    while(i<x-1){
        i=i+1;
        sob=x%i;
        if (sob==0){
            return 0;
        }
        if (i==x-1){
            return x;
        }
    }
}
// funcion principal
int main(){
    int x=0,y=0,ret=0;
    cout << "digite inicio de rango: "; cin >> x;
    cout << endl;
    cout << "digite final de rango: "; cin >> y;
    cout << endl;
    int aux = x - 1;
//ciclo en donde se examina que un numero dentro del rango no sea divisible mas que por 1 y el mismo
    while(aux<y){
        salto:
        aux=aux+1;
        ret=primus(aux);
        //parametro de salida para una sola linea
        if (aux==x){
            cout << "los numeros primos en el rango indicado son " << endl;
        }
        //parametro que borra los return de 0 de el resultado final
        if (ret==0){
            goto salto;
        }
        //parametro que elimina un error de conteo donde muestra un numero primo mayor al del final de rango
        if (ret > y){
            return 0;
        }
        cout << ret;
        cout << ", ";
    }
    return 0;
}

